### PR TITLE
feat(web): add xcode-cloud env-vars list/set/delete commands

### LINKED ISSUE #817: feat(web): Xcode Cloud environment variable management
## Summary

Add support for managing Xcode Cloud environment variables (list, set, delete) via `asc web xcode-cloud env-vars`. This covers both **workflow-level** and **product-level** environment variables, including **secret (encrypted)** variables.

The App Store Connect REST API does not expose environment variable endpoints. This feature uses Apple's private CI API (`/ci/api`), discovered via browser network inspection of the ASC web UI.

---

## Private API Endpoints

All endpoints live under `/ci/api` and reuse the existing web session auth (same as the usage reporting endpoints).

| Method | Path | Purpose |
|--------|------|---------|
| `GET` | `/teams/{teamId}/products/{productId}/workflows-v15/{workflowId}` | Get single workflow (includes `environment_variables`) |
| `GET` | `/teams/{teamId}/products/{productId}/workflows-v15?limit=100&include_deleted=false` | List all workflows |
| `PUT` | `/teams/{teamId}/products/{productId}/workflows-v15/{workflowId}` | Update workflow (full body replace, including env vars) |
| `GET` | `/teams/{teamId}/products/{productId}/product-environment-variables` | Get product-level shared env vars |
| `GET` | `/ci/auth/keys/client-encryption` | Get P-256 public key for secret encryption |

### Key observations

- **No dedicated env var CRUD endpoints** — env vars are embedded in the workflow body. Adding/updating/deleting means GET the workflow, modify the `environment_variables` array, and PUT the full workflow back.
- **Two levels of env vars**:
  - **Workflow-level**: per-workflow, stored in `environment_variables` array
  - **Product-level**: shared across workflows, managed via separate endpoint, referenced by UUID in `product_environment_variables` array
- PUT requires the **full workflow body** (name, description, start_conditions, versions, actions, post_actions, disabled, locked, clean, environment_variables, product_environment_variables, container_file_path, repo).
- Client-generated UUIDs for new env var IDs are replaced by server-assigned UUIDs in the response.
- **`x-apple-signature` header is NOT required** — tested and confirmed that PUT requests succeed with session cookies alone.

---

### Three value shapes

| Direction | Plaintext | Secret |
|-----------|-----------|--------|
| Client → Server | `{"plaintext": "value"}` | `{"ciphertext": "base64..."}` |
| Server → Client | `{"plaintext": "value"}` | `{"redacted_value": ""}` |

### Workflow-level env var (in workflow content)

```json
{
  "id": "uuid",
  "name": "VAR_NAME",
  "value": { "plaintext": "value" }
}
```

### Product-level env var (from `/product-environment-variables`)

```json
{
  "id": "uuid",
  "name": "SHARED_API_TOKEN",
  "value": { "redacted_value": "" },
  "is_locked": true,
  "related_workflow_summaries": [
    {
      "id": "uuid",
      "name": "TestFlight Deploy",
      "disabled": false,
      "locked": false,
      "last_modified_by": "System",
      "last_modified_at": "2026-02-26T07:27:57.477Z"
    }
  ]
}
```

Product-level vars:
- Are always secret (redacted)
- Have an `is_locked` field
- Include `related_workflow_summaries` showing which workflows reference them

---

## Secret Encryption (ECIES with HKDF + AES-256-GCM)

When adding a **secret** environment variable, the value must be encrypted client-side before sending. The algorithm was reverse-engineered from the ASC web UI's minified JavaScript (`main.*.js`).

### Algorithm

```
1. GET /ci/auth/keys/client-encryption
   → {"key": "<base64-encoded P-256 public key, 64 bytes raw (x || y)>"}

2. Prepend 0x04 byte to server key → 65-byte uncompressed EC point
   Import as ECDH P-256 public key

3. Generate ephemeral ECDH P-256 key pair

4. ECDH key agreement:
   shared_secret = ECDH(ephemeral_private, server_public)
   → raw 256-bit key (AES-256-GCM format)

5. Re-import shared_secret as HKDF base key

6. salt = random 32 bytes

7. HKDF-SHA256:
   final_key = HKDF(key=shared_secret, salt=salt, info=b"")
   → AES-256-GCM 256-bit key

8. iv = random 12 bytes

9. AES-256-GCM encrypt:
   ciphertext_with_tag = AES-GCM(key=final_key, iv=iv, plaintext=UTF8(value))

10. Concatenate output:
    output = salt(32) || ephemeral_pubkey_no_prefix(64) || iv(12) || ciphertext_with_tag

11. Base64(output) → the "ciphertext" field in the request body
```

### Output size

`124 + len(plaintext)` bytes before base64 encoding (32 salt + 64 ephemeral key + 12 IV + 16 GCM tag + plaintext length).

### Validation

**The Go implementation has been validated end-to-end against the live API:**

1. Implemented ECIES in Go using `crypto/ecdh` (P-256), `crypto/aes` + `crypto/cipher` (AES-256-GCM), `golang.org/x/crypto/hkdf` (HKDF-SHA256)
2. Encrypted a test value ("encrypted-by-go") using the Go implementation
3. Sent the Go-encrypted ciphertext to the live ASC API via PUT
4. **Server returned 200** and the variable appeared as `{"redacted_value": ""}` — confirming the Go encryption is accepted

Test results:
| Test | Result |
|------|--------|
| GET workflow (read env vars) | ✅ 200 |
| GET encryption key | ✅ 200 |
| Go ECIES encrypt → correct output structure | ✅ Pass |
| Go ECIES encrypt → different each time (randomized) | ✅ Pass |
| PUT with Go-encrypted secret | ✅ 200 — server accepted |
| PUT with plaintext var | ✅ 200 — value preserved |
| PUT to delete vars (omit from array) | ✅ 200 |
| `x-apple-signature` required? | ❌ Not required |

### Confidence level

- **Algorithm**: Very high — extracted verbatim from Apple's JS source, validated with live API.
- **Go implementation**: Very high — tested end-to-end, server accepted Go-encrypted ciphertexts.
- **Per-account variation**: The algorithm is identical for all accounts (same static JS bundle). The server public key may rotate per team or periodically — always fetch fresh before encrypting.
- **Inputs**: Only the plaintext value string is external. Server key comes from the API. Salt, IV, and ephemeral key are generated fresh each time.

### Reference JS (minified, from ASC web UI)

```javascript
n = await this.getPublicKey();
r = E => { P=atob(E); j=P.length; X=new Uint8Array(j+1); X[0]=4;
            for(re=0;re<j;re++) X[re+1]=P.charCodeAt(re); return X.buffer };
i = await crypto.subtle.importKey("raw",r(n),{name:"ECDH",namedCurve:"P-256"},!1,[]);
o = await crypto.subtle.generateKey({name:"ECDH",namedCurve:"P-256"},!1,["deriveKey","deriveBits"]);
c = await crypto.subtle.deriveKey({name:"ECDH",public:i},o.privateKey,
    {name:"AES-GCM",length:256},!0,["encrypt","decrypt"]);
l = await crypto.subtle.importKey("raw",await crypto.subtle.exportKey("raw",c),
    "HKDF",!1,["deriveKey"]);
u = crypto.getRandomValues(new Uint8Array(32));
s = await crypto.subtle.deriveKey({name:"HKDF",hash:"SHA-256",salt:u,
    info:new TextEncoder().encode("")},l,{name:"AES-GCM",length:256},!0,["encrypt","decrypt"]);
m = crypto.getRandomValues(new Uint8Array(12));
g = await crypto.subtle.encrypt({name:"AES-GCM",iv:m},s,new TextEncoder().encode(t));
C = await crypto.subtle.exportKey("raw",o.publicKey);
T = [u, new Uint8Array(C.slice(1)), m, new Uint8Array(g)];
// → concatenate T arrays, base64 encode → {"ciphertext": result}
```

---

## Workflow PUT Body Shape

The full workflow body must be sent on PUT (not just the env vars):

```json
{
  "name": "PR Build",
  "description": "...",
  "start_conditions": {
    "manual": { "branch": { "branch": { "kind": "branch", "is_all_match": true, "patterns": [] } } }
  },
  "macos_version": { "build": "latest:stable", "name": "Latest Release" },
  "xcode_version": { "build": "latest:stable", "name": "Latest Release" },
  "actions": [{ "id": "...", "default_name": "...", "action_type": "archive", ... }],
  "post_actions": [],
  "disabled": false,
  "locked": false,
  "clean": true,
  "environment_variables": [
    { "id": "uuid", "name": "VAR", "value": { "plaintext": "val" } },
    { "id": "uuid", "name": "SECRET", "value": { "ciphertext": "base64..." } }
  ],
  "product_environment_variables": ["uuid-refs"],
  "container_file_path": "App.xcodeproj",
  "repo": "uuid"
}
```

### PUT Response

```json
{
  "id": "workflow-uuid",
  "content": { /* same shape as request, with server-assigned env var IDs */ },
  "metadata": {
    "is_deleted": false,
    "last_modified_by": "Name",
    "last_modified_at": "2026-03-01T07:31:46.525Z"
  }
}
```

---

## Proposed CLI Commands

```bash
# List workflow env vars
asc web xcode-cloud env-vars list --product-id ID --workflow-id ID

# List product-level env vars
asc web xcode-cloud env-vars list --product-id ID --product-level

# Set a plaintext env var on a workflow
asc web xcode-cloud env-vars set --product-id ID --workflow-id ID --name KEY --value VALUE

# Set a secret env var on a workflow
asc web xcode-cloud env-vars set --product-id ID --workflow-id ID --name KEY --value VALUE --secret

# Delete an env var from a workflow
asc web xcode-cloud env-vars delete --product-id ID --workflow-id ID --name KEY
```

---

### Existing code to extend
- `internal/web/ci.go` — Add `GetCIWorkflow`, `UpdateCIWorkflow`, `GetProductEnvironmentVariables`, `GetClientEncryptionKey` methods. Expand `CIWorkflowContent` type.
- `internal/cli/web/web_xcode_cloud.go` — Add `env-vars` subcommand group.

### Encryption in Go
Standard library support: `crypto/ecdh` (P-256), `crypto/aes` + `crypto/cipher` (AES-256-GCM), `golang.org/x/crypto/hkdf` (HKDF-SHA256). No new external dependencies needed — `golang.org/x/crypto` is already in `go.mod`.

### Prototype
A working Go ECIES implementation exists in `internal/web/ci_encrypt_test.go` (validated against live API).

### Testing
- Unit tests for encryption (encrypt with known key, verify output structure and size) — already passing
- httptest-based tests for API client methods
- cmdtest-based CLI integration tests

### PR DESCRIPTION
**Pull request by mithileshchellappan**
## Summary

- Add `asc web xcode-cloud env-vars list|set|delete` commands to manage Xcode Cloud workflow environment variables via Apple's private CI API
- Support plaintext and ECIES-encrypted secret variables (same encryption scheme as the ASC web UI)
- Implement GET/PUT round-trip for workflows using `json.RawMessage` to preserve unknown fields
- 25 new tests (16 CLI-level + 9 API-layer) covering success paths, error handling, flag validation, and edge cases

### Commands
```
asc web xcode-cloud env-vars list   --product-id ID --workflow-id ID [--output table|json|markdown]
asc web xcode-cloud env-vars set    --product-id ID --workflow-id ID --name NAME --value VALUE [--secret]
asc web xcode-cloud env-vars delete --product-id ID --workflow-id ID --name NAME
```

### Files changed
| File | Action | Purpose |
|------|--------|---------|
| `internal/web/ci_encrypt.go` | New | ECIES encryption (P-256 ECDH + HKDF-SHA256 + AES-256-GCM) |
| `internal/web/ci_encrypt_test.go` | New | Encryption output structure, non-determinism, and live-test helper |
| `internal/web/ci.go` | Edit | Add types + methods: `GetCIWorkflow`, `UpdateCIWorkflow`, `GetCIEncryptionKey`, `ExtractEnvVars`, `SetEnvVars` |
| `internal/web/ci_test.go` | Edit | 9 new tests for API methods and env var helpers |
| `internal/cli/web/web_xcode_cloud_envvars.go` | New | CLI commands: list, set, delete |
| `internal/cli/web/web_xcode_cloud_envvars_test.go` | New | 16 CLI-level tests |
| `internal/cli/web/web_xcode_cloud.go` | Edit | Register `env-vars` subcommand |
| `internal/cli/web/web_xcode_cloud_test.go` | Edit | Update hierarchy assertion (2 → 3 subcommands) |

## Test plan

- [x] `ASC_BYPASS_KEYCHAIN=1 make test` — all tests pass
- [x] Live smoke test: `env-vars list` returns workflow variables
- [x] Live smoke test: `env-vars set` creates plaintext variable
- [x] Live smoke test: `env-vars set --secret` creates encrypted variable
- [x] Live smoke test: `env-vars delete` removes variable
- [x] PUT body sends raw content (not wrapped), validated by live 200 response
- [x] Upsert preserves existing variable ID on update
- [x] Unknown workflow content fields survive round-trip

Closes #817

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/web/ci.go [hunk 1]
```diff
@@ -342,3 +369,105 @@ func (c *Client) ListCIWorkflows(ctx context.Context, teamID, productID string)
...
+	}
+	path := "/teams/" + url.PathEscape(teamID) + "/products/" + url.PathEscape(productID) + "/workflows-v15/" + url.PathEscape(workflowID)
+	_, err := c.doRequest(ctx, "PUT", path, content)
+	return err
+}
+
+// GetCIEncryptionKey fetches the P-256 public key for secret encryption.
+// GET /ci/auth/keys/client-encryption
+func (c *Client) GetCIEncryptionKey(ctx context.Context) (*CIEncryptionKeyResponse, error) {
+	body, err := c.doRequest(ctx, "GET", "/ci/auth/keys/client-encryption", nil)
```
[the remark this answers is not in the document]
Comment (mithileshchellappan) [reply internal/web/ci.go]: Already fixed in e4b231f. The comment misreads the code — it claims the path is `/ci/auth/keys/client-encryption`, but the actual code uses `/auth/keys/client-encryption` (relative to the `/ci/api` base URL), producing the correct full path. The comment text and the fix commit message even match. Live smoke tests with `--secret` confirmed 200 responses.

File: internal/web/ci.go [hunk 2]
```diff
@@ -342,3 +369,108 @@ func (c *Client) ListCIWorkflows(ctx context.Context, teamID, productID string)
...
+// ExtractEnvVars extracts environment_variables from raw workflow content.
+func ExtractEnvVars(content json.RawMessage) ([]CIEnvironmentVariable, error) {
+	var m map[string]json.RawMessage
+	if err := json.Unmarshal(content, &m); err != nil {
+		return nil, fmt.Errorf("failed to decode workflow content: %w", err)
+	}
+	raw, ok := m["environment_variables"]
+	if !ok {
+		return nil, nil
+	}
```
[the remark this answers is not in the document]
Comment (mithileshchellappan) [reply internal/web/ci.go]: Theoretical — the API never returns null content for a valid workflow GET. The `json.Unmarshal` on line 534 would already return an error for malformed input (e.g. non-JSON). For the missing-key case, returning `nil, nil` is intentional: it means "no env vars key present" which is a valid state for workflows that have never had env vars configured. Callers handle this as an empty list.

**Comment by rudrankriyam** [comment]
⚡️
